pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -446,10 +485,19 @@ func (nc *Controller) doNoScheduleTaintingPassWorker() {
 			return
 		}
 		nodeName := obj.(string)
-
+		needsToRequeue := false
 		if err := nc.doNoScheduleTaintingPass(nodeName); err != nil {
-			// TODO (k82cn): Add nodeName back to the queue.
 			klog.Errorf("Failed to taint NoSchedule on node <%s>, requeue it: %v", nodeName, err)
+			needsToRequeue = true
+		}
+		if err := nc.reconcileNodeLables(nodeName); err != nil {
+			klog.Errorf("Failed to reconcile labels for node <%s>, requeue it: %v", nodeName, err)
+			needsToRequeue = true
+		}
+
+		// Requeue the node if needed.
+		if needsToRequeue {
+			nc.nodeUpdateQueue.Add(nodeName)
```

Comment:
If add it back to queue, it's better to use working queue with ratelimit.

Reply:
Done.

Reply:
For ratelimit working queue, we need to use `AddRateLimited` and call `Forget(req)`  if succeed.

Reply:
You're right. Sorry for missing that. Fixed in the new commit